### ISSUE TITLE
Ubuntu RDP 22.04.01 out of the box is not working

### ISSUE THREAD

**Issue by maxima120** [FreeRDP/FreeRDP#8159]
I cannot find a suitable name for this because there is bunch of FreeRDP errors. I dont know which one is causing it in particular. So please bear with me..

# How to repro:

Installed `Ubuntu 22.04.01` as Hyper-V VM on host OS `Windows 11 Pro x64` 
Updated it by `sudo apt upgrade`. 
Enabled Settings-Sharing, RDP, set password
Tried to connect from Windows 11 and received "Because of protocol error this session will be disconnected... Error code 0xd06"
ran `systemctl --user status gnome-remote-desktop` and although the status is green there are errors. All of them are pointing to FreeRDP.

# Here is extract from journalctl (I think meaningful info is starting at 21:21:51):

```
Aug 29 21:20:34 ub-tf gnome-remote-de[16488]: RDP server started
Aug 29 21:21:00 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:665] [16488:16553] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
Aug 29 21:21:00 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
Aug 29 21:21:01 ub-tf gnome-remote-de[16488]: [RDP.RDPGFX] CapsAdvertise: Accepting capability set with version RDPGFX_CAPVERSION_106, Client cap flags: H264 (AVC444): true, H264 (AVC420): true
Aug 29 21:21:01 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:01:692] [16488:16553] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
Aug 29 21:21:01 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:01:692] [16488:16553] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
Aug 29 21:21:01 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:01:692] [16488:16557] [ERROR][com.freerdp.channels.rdpgfx.server] - WTSVirtualChannelRead failed!
Aug 29 21:21:01 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:01:692] [16488:16557] [ERROR][com.freerdp.channels.rdpgfx.server] - rdpgfx_server_handle_messages failed with error 1359
Aug 29 21:21:01 ub-tf gnome-remote-de[16488]: Unable to check file descriptor, closing connection
Aug 29 21:21:01 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:01:695] [16488:16488] [ERROR][com.freerdp.core.transport] - BIO_should_retry returned a system error 32: Broken pipe
Aug 29 21:21:51 ub-tf gnome-remote-de[16488]: RDP server stopped
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:20:57:754] [16488:16553] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:20:57:754] [16488:16553] [INFO][com.freerdp.core.connection] - Server Security: NLA:1 TLS:0 RDP:0
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:20:57:754] [16488:16553] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:1 TLS:0 RDP:0
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] - Accepted client: XXXXX
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] - Accepted channels:
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] -  rdpdr
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] -  rdpsnd
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] -  cliprdr
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.connection] -  drdynvc
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
Aug 29 21:21:51 ub-tf gnome-remote-desktop-daemon[16488]: [21:21:00:666] [16488:16553] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
Aug 29 21:21:52 ub-tf gnome-remote-de[16588]: RDP server started
Aug 29 21:22:02 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:101] [16588:16618] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
Aug 29 21:22:02 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
Aug 29 21:22:02 ub-tf gnome-remote-de[16588]: [RDP.RDPGFX] CapsAdvertise: Accepting capability set with version RDPGFX_CAPVERSION_106, Client cap flags: H264 (AVC444): true, H264 (AVC420): true
Aug 29 21:22:02 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:124] [16588:16618] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
Aug 29 21:22:02 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:124] [16588:16618] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
Aug 29 21:22:02 ub-tf gnome-remote-de[16588]: Unable to check file descriptor, closing connection
Aug 29 21:22:02 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:127] [16588:16588] [ERROR][com.freerdp.core.transport] - BIO_should_retry returned a system error 32: Broken pipe
Aug 29 21:22:15 ub-tf gnome-remote-de[16588]: RDP server stopped
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:21:58:892] [16588:16618] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:21:58:892] [16588:16618] [INFO][com.freerdp.core.connection] - Server Security: NLA:1 TLS:0 RDP:0
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:21:58:892] [16588:16618] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:1 TLS:0 RDP:0
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] - Accepted client: XXXXX
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] - Accepted channels:
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] -  rdpdr
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] -  rdpsnd
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] -  cliprdr
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.connection] -  drdynvc
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
Aug 29 21:22:15 ub-tf gnome-remote-desktop-daemon[16588]: [21:22:02:102] [16588:16618] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
Aug 29 21:23:32 ub-tf gnome-remote-de[1824]: RDP server started
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:153] [1824:2525] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:154] [1824:2525] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
Aug 29 21:23:44 ub-tf gnome-remote-de[1824]: [RDP.RDPGFX] CapsAdvertise: Accepting capability set with version RDPGFX_CAPVERSION_106, Client cap flags: H264 (AVC444): true, H264 (AVC420): true
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:177] [1824:2525] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:177] [1824:2525] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:177] [1824:2536] [ERROR][com.freerdp.channels.rdpgfx.server] - WTSVirtualChannelRead failed!
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:177] [1824:2536] [ERROR][com.freerdp.channels.rdpgfx.server] - rdpgfx_server_handle_messages failed with error 1359
Aug 29 21:23:44 ub-tf gnome-remote-de[1824]: Unable to check file descriptor, closing connection
Aug 29 21:23:44 ub-tf gnome-remote-desktop-daemon[1824]: [21:23:44:181] [1824:1824] [ERROR][com.freerdp.core.transport] - BIO_should_retry returned a system error 32: Broken pipe
Aug 29 21:24:12 ub-tf gnome-remote-desktop-daemon[1824]: [21:24:12:647] [1824:2662] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
Aug 29 21:24:12 ub-tf gnome-remote-desktop-daemon[1824]: [21:24:12:648] [1824:2662] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
Aug 29 21:24:12 ub-tf gnome-remote-de[1824]: [RDP.RDPGFX] CapsAdvertise: Accepting capability set with version RDPGFX_CAPVERSION_106, Client cap flags: H264 (AVC444): true, H264 (AVC420): true
Aug 29 21:24:12 ub-tf gnome-remote-desktop-daemon[1824]: [21:24:12:675] [1824:2662] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
Aug 29 21:24:12 ub-tf gnome-remote-desktop-daemon[1824]: [21:24:12:675] [1824:2662] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
Aug 29 21:24:12 ub-tf gnome-remote-de[1824]: Unable to check file descriptor, closing connection
Aug 29 21:24:12 ub-tf gnome-remote-desktop-daemon[1824]: [21:24:12:680] [1824:1824] [ERROR][com.freerdp.core.transport] - BIO_should_retry returned a system error 32: Broken pipe

```
![image](https://user-images.githubusercontent.com/10121753/187286854-f15e5d8b-ff94-416d-8325-daefbd73971e.png)

**Comment by mfleisz** [comment]
`Connection reset by peerConnection reset by peer` is usually an indication that the server cut the connection due to whatever reason. 

Have you tried connecting with mstsc to the same server with the same user? Does that work?

**Comment by maxima120** [comment]
apparently there is a bug in Windows RDP client. You need to **change display colours to 32-bit** - tested, working.

![image](https://user-images.githubusercontent.com/10121753/187432541-ae07feb4-24fe-4a40-b9c7-a85edd1fc961.png)

https://askubuntu.com/questions/1426376/native-rdp-on-ubuntu-22-04-not-working/1426453#1426453

**Comment by deliciouslytyped** [comment]
Setting it to 32bit fixed my similar issue with MobaXterm 23.1, thanks for documenting this!
